src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -104,6 +105,18 @@ public sealed class EmitOptions : IEquatable<EmitOptions>
         /// </summary>
         public string? RuntimeMetadataVersion { get; private set; }
 
+        /// <summary>
+        /// The default source encoding used to parse source before producing a compilation. Only embedded in
+        /// the pdb if the value is non-null. 
+        /// </summary>
+        public Encoding? DefaultSourceFileEncoding { get; private set; }
+
+        /// <summary>
+        /// The fallback source encoding used to parse source before producing a compilation. Only embedded
+        /// in the pdb if the value is non-null.
+        /// </summary>
+        public Encoding? FallbackSourceFileEncoding { get; private set; }
```

Comment:
Under what circumstances will this be `null`? Particularly when the above is non-null?

Reply:
This will be null if defaultEncoding is provided (this comes from the -codePage), or UTF8 was used

Reply:
Also null if BOM for all files is used. This comment on defaultEncoding for SourceText.Create explains it slightly better I think: 

```
/// <param name="defaultEncoding">
        /// Specifies an encoding to be used if the actual encoding can't be determined from the stream content (the stream doesn't start with Byte Order Mark).
        /// If not specified auto-detect heuristics are used to determine the encoding. If these heuristics fail the decoding is assumed to be Encoding.Default.
        /// Note that if the stream starts with Byte Order Mark the value of <paramref name="defaultEncoding"/> is ignored.
        /// </param>
```

`BOM > default encoding > UTF8 > Detected` seems to be the order. If ever detected is triggered, it needs to be recorded what that value was